hazelcast-client/src/main/java/com/hazelcast/client/spi/properties/ClientProperty.java

```diff
@@ -167,6 +167,30 @@
     public static final HazelcastProperty IO_BALANCER_INTERVAL_SECONDS
             = new HazelcastProperty("hazelcast.client.io.balancer.interval.seconds", 20, SECONDS);
 
+    /**
+     * Optimization that allows sending of packets over the network to be done on the calling thread if the
+     * conditions are right. This can reduce latency and increase performance for low threaded environments.
+     *
+     * It is disabled.
+     */
+    public static final HazelcastProperty IO_WRITE_THROUGH_ENABLED
+            = new HazelcastProperty("hazelcast.client.io.write.through", false);
+
+    /**
+     * Property needed for concurrency detection so that write through and dynamic response handling
+     * can be done correctly. This property sets the window the concurrency detection will signalling
+     * that concurrency has been detected, even if there are no further updates in that window.
+     *
+     * Normally in a concurrency system the windows keeps sliding forward so it will always remain
+     * concurrent.
+     *
+     * Setting it too high effectively disabled the optimization because once concurrency has been detected
+     * it will keep that way. Setting it too low could lead to suboptimal performance because the system
+     * will try write through and other optimization even though the system is concurrent.
+     */
+    public static final HazelcastProperty CONCURRENT_WINDOW_MS
```

Comment:
Minor comment: maybe this is a too specific detail to be tweaking. It would be good if we had a value which didn't need changing. Alas, as you say, there is no right answer.

One thing that you might change - establish a better link between this property and `IO_WRITE_THROUGH_ENABLED` by adding something like `Needs {@link #IO_WRITE_THROUGH_ENABLED} to be set to {@code true}.` and `@see #CONCURRENT_WINDOW_MS` on the other property.

Reply:
One more thing - now I see this has an effect even if `IO_WRITE_THROUGH_ENABLED` is disabled but `BACKPRESSURE_ENABLED` is true. Is this intentional? Can you clarify?

Reply:
The idea is that in 3.13 we'll enable these features by default so that they can be turned of if needed but they will be on by default.